### PR TITLE
Add doctests for fractional knapsack

### PR DESCRIPTION
**Pull request by Hardvan**
Added doctests for the Fractional Knapsack algorithm, as requested in PR #10888 

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: greedy_methods/fractional_knapsack.py [hunk 1]
```diff
@@ -6,6 +6,10 @@ def frac_knapsack(vl, wt, w, n):
     """
     >>> frac_knapsack([60, 100, 120], [10, 20, 30], 50, 3)
     240.0
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 10, 4)
+    105.0
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 4)
+    95.0
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:12]:
```suggestion
    95.0
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6], 8, 4)
    ?
    >>> frac_knapsack([10, 40, 30], [5, 4, 6, 3], 8, 4)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 0, 4)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 0)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], -8, 4)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, -4)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 800, 4)
    ?
    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 400)
    ?
```

**Comment by cclauss** [comment]
On your computer do `python3 -m doctest -v filename.py` to run the tests.

**Comment by Hardvan** [comment]
I have successfully executed the doctests, yielding the following output:

11 tests in fractional_knapsack.frac_knapsack
11 tests in 2 items.
11 passed and 0 failed.
Test passed.

File: greedy_methods/fractional_knapsack.py (hunk 1 above)
Comment (Hardvan) [reply greedy_methods/fractional_knapsack.py:12]: I have added the above doctests to the code.

File: greedy_methods/fractional_knapsack.py [hunk 2]
```diff
@@ -11,21 +11,21 @@ def frac_knapsack(vl, wt, w, n):
...
+    0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, -4)
-    ?
+    95.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 800, 4)
-    ?
+    130
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 400)
-    ?
+    95.0
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:28]: Just one more... I promise.
```suggestion
    95.0
    >>> frac_knapsack("ABCD", [5, 4, 6, 3], 8, 400)
    ?
```

**Comment by Hardvan** [comment]
After adding this test case, this is the output:
r = sorted(zip(vl, wt), key=lambda x: x[0] / x[1], reverse=True)
    TypeError: unsupported operand type(s) for /: 'str' and 'int'
**********************************************************************
1 items had failures:
   1 of  12 in __main__.frac_knapsack
***Test Failed*** 1 failures.

File: greedy_methods/fractional_knapsack.py [hunk 3]
```diff
@@ -6,6 +6,28 @@ def frac_knapsack(vl, wt, w, n):
...
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], -8, 4)
+    0
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, -4)
+    95.0
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 800, 4)
+    130
+    >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 400)
+    95.0
+    >>> frac_knapsack("ABCD", [5, 4, 6, 3], 8, 400)
+    ?
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:30]:
```suggestion
    Traceback (most recent call last):
        ...
    TypeError: unsupported operand type(s) for /: 'str' and 'int'
```

**Pull request merged**